frontend/src/app/app.tsx

```diff
@@ -1,23 +1,34 @@
+import { actions as authActions } from '~/bundles/auth/store';
 import { Link, RouterOutlet } from '~/bundles/common/components/components.js';
 import { AppRoute } from '~/bundles/common/enums/enums.js';
 import {
     useAppDispatch,
+    useAppSelector,
     useEffect,
     useLocation,
 } from '~/bundles/common/hooks/hooks.js';
 import { actions as userActions } from '~/bundles/users/store';
+import { storage, StorageKey } from '~/framework/storage/storage';
 
 const App: React.FC = () => {
     const { pathname } = useLocation();
+    const { user } = useAppSelector((state) => state.auth);
     const dispatch = useAppDispatch();
+    const token = storage.getSync(StorageKey.TOKEN);
 
     const isRoot = pathname === AppRoute.ROOT;
 
     useEffect(() => {
         if (isRoot) {
             void dispatch(userActions.loadAll());
         }
-    }, [isRoot, dispatch]);
+    }, [dispatch, isRoot]);
+
+    useEffect(() => {
+        if (!user && token) {
```

Comment:
we don't care about token. If we don't have token we will get error

Reply:
Resolved in [62a6191](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/120/commits/62a6191e0a0cc77a168d461945b777155b7ef161)